src/psi/j/executors/saga.py

```diff
@@ -0,0 +1,222 @@
+import atexit
+import logging
+from threading import RLock
+from typing import Optional, Dict, List
+
+import radical.saga as rs
+import radical.saga.job.constants as ct
+import time
+
+from psi.j import JobExecutor, JobExecutorConfig, JobState, JobStatus, Job, \
+    SubmitException, InvalidJobException
+
+logger = logging.getLogger(__name__)
+
+
+# map SAGA states to jpsi states.
+_STATE_MAP = {
+    rs.job.NEW: JobState.NEW,
+    rs.job.PENDING: JobState.QUEUED,
+    rs.job.RUNNING: JobState.ACTIVE,
+    rs.job.DONE: JobState.COMPLETED,
+    rs.job.FAILED: JobState.FAILED,
+    rs.job.CANCELED: JobState.CANCELED,
+}
+
+
+class _JobMapping:
+    def __init__(self, jpsi_job: Job, saga_job: Optional[rs.job.Job] = None):
+        self.jpsi_job = jpsi_job
+        self.saga_job = saga_job
+
+
+class SagaExecutor(JobExecutor):
+    """
+    An executor that use SAGA to run jobs.
+
+    For details about SAGA, see https://en.wikipedia.org/wiki/Simple_API_for_Grid_Applications
+    and related links.
+    """
+
+    assert('job' in dir(rs)), 'incomplete installation'
```

Comment:
How can that happen?

Reply:
That was there in the original version. I tried to make minimal changes.

So if you think it should be removed, it should probably be removed.

Reply:
Ha! :-)  I think this can be removed by now.